pkg/repos/repo_collection.go

```diff
@@ -49,21 +49,19 @@ type repoCollection struct {
 //
 // fsys: base file system (used for unit testing)
 // specs: a list of repo.Spec entries (usually from the config file)
-// workdirSupport: true if the provided fsys supports workdirs
 //
 // **Returns:**
 //
 // RepoCollection: assembled RepoCollection, or nil if there was an error
 // error: an error if there is a problem
-func NewRepoCollection(fsys afero.Fs, specs []Spec, workdirSupport bool) (RepoCollection, error) {
+func NewRepoCollection(fsys afero.Fs, specs []Spec, basePath string) (RepoCollection, error) {
```

Comment:
`basePath` parameter added to simplify code and close several classes of bugs related to absPath handling